### PR TITLE
Task/cdd 2169 new landing page cms

### PR DESCRIPTION
**Pull request by phill-stanley**
# Description

This PR includes CMS updates for the new `LandingPage`

- New landing page model for the CMS
- Updates to build_cms_site command to create initial landing page

Fixes #CDD-2169

---

## Type of change

Please select the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Tech debt item (this is focused solely on addressing any relevant technical debt)

---

# Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests at the right levels to prove my change is effective
- [ ] I have added screenshots or screen grabs where appropriate
- [x] I have added docstrings in the correct style [(google)](https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings)

### CODE REVIEW

File: cms/landing_page/apps.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+from django.apps import AppConfig
```
Comment (A-Ashiq) [inline cms/landing_page/apps.py:1]: What was your reasoning behind making an entirely new wagtail app for this?
What stopped you from just making a new `LandingPage` model within the home app? That way it would be referred to as `home.LandingPage` instead of `landing_page.LandingPage`

File: cms/landing_page/models.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+class LandingPage(UKHSAPage):
+    is_creatable = False
+    max_count = 1
+    sub_title = models.CharField(max_length=255)
+    body = ALLOWABLE_BODY_CONTENT_SECTION_LINK
+
+    content_panels = Page.content_panels + [FieldPanel("sub_title"), FieldPanel("body")]
+
+    api_fields = [
```
Comment (A-Ashiq) [inline cms/landing_page/models.py:19]: If you add the list from the base page, you don't need to repeat the seo fields here:
```
    api_fields = UKHSAPage.api_fields + [
        APIField("title"),
        APIField("sub_title"),
        APIField("body")
    ]
```

File: cms/landing_page/models.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    )
+
+    objects = LandingPageManager()
+
+    @classmethod
+    def is_previewable(cls):
+        """Returns False. This is a headline CMS, preview panel is not supported ."""
+        return False
+
+    def get_url_parts(self, request=None) -> tuple[int, str, str]:
```
Comment (A-Ashiq) [inline cms/landing_page/models.py:45]: Why did you need this here? This is inherited from the `UKHSAPage` model

File: cms/landing_page/models.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+        Returns:
+            Tuple containing the URL parts:
+                1) ID of the corresponding `Site` record
+                2) The root URL of the site
+                    e.g. `https://ukhsa-dashboard.data.gov.uk`
+                3) The path of the current page.
+                    This always returns an empty string
+                    e.g. `""`
+                    This is because the homepage is assumed
```
Comment (A-Ashiq) [inline cms/landing_page/models.py:69]:
```suggestion
                    This is because the landing page is assumed
```

File: cms/dashboard/management/commands/build_cms_site_helpers/__init__.py [hunk 5]
```diff
@@ -4,6 +4,7 @@
     create_composite_page,
     create_bulk_downloads_page,
     create_landing_dashboard_page,
```
Comment (A-Ashiq) [inline cms/dashboard/management/commands/build_cms_site_helpers/__init__.py:6]: Might be worth renaming this old handler ot be `create_home_page`, so its a bit clearer.
That's my bad, I saw my name against the git blame for that. Which in my defence was way before we embarked on the actual landing page re-work 😄

File: cms/dashboard/management/commands/build_cms_site_helpers/landing_page.py [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+        slug="other-respiratory-viruses"
+    )
+    weather_health_alerts_page = CompositePage.objects.get(slug="weather-health-alerts")
+
+    return [
+        {
+            "type": "section",
+            "value": {
+                "heading": "Respiratory viruses",
+                "page_link": None,
```
Comment (A-Ashiq) [inline cms/dashboard/management/commands/build_cms_site_helpers/landing_page.py:18]: Presumably we're gonna have to come back and slide the links in for the topic index pages here?

**Comment by A-Ashiq** [comment]
Screenshot 2024-09-26 at 10 59 52

You'll need to make differnent help text for this? We restricted to 2 cards in the old home page, but we'll have a column limit this time around?

File: cms/dashboard/management/commands/build_cms_site_helpers/landing_page.py (hunk 6 above)
Comment (phill-stanley) [reply cms/dashboard/management/commands/build_cms_site_helpers/landing_page.py:18]: Yeah, these will be linked to the index pages when they're in. On the front-end these fields are nullable and will fallback to a dark heading rather than the blue link style.

File: cms/dashboard/management/commands/build_cms_site_helpers/__init__.py (hunk 5 above)
Comment (phill-stanley) [reply cms/dashboard/management/commands/build_cms_site_helpers/__init__.py:6]: yeah i'll update the naming here when I move landing_page to home app.

File: cms/landing_page/apps.py (hunk 1 above)
Comment (phill-stanley) [reply cms/landing_page/apps.py:1]: From our conversation I'll refactor this into the `Home` app.

File: cms/home/managers/landingpage.py [hunk 7]
```diff
@@ -0,0 +1,35 @@
+from django.db import models
```
Comment (A-Ashiq) [inline cms/home/managers/landingpage.py:1]: Nitpick:
Can you rename these files as `home_page.py` instead of `homepage.py`.
I fully appreciate some people prefer the latter but it feels needlessly unreadable

**Review by A-Ashiq — APPROVED** [review]
